test/EngineTest.cpp

```diff
@@ -43,94 +57,259 @@ IdTable makeIdTableFromVector(std::vector<std::array<size_t, TABLE_WIDTH>> table
+      traceMessage << table(row, column) << " ";
+    }
+    traceMessage << "\n";
+  }
+  traceMessage << "with IdTable described by vector\n";
+  for (auto const& row: sampleSolution) {
+    for (auto const& entry: row) {
+      traceMessage << entry << " ";
+    }
+    traceMessage << "\n";
+  }
+  auto trace = generateLocationTrace(T, traceMessage.str());
+  
+  // Do the IdTable and sampleSolution have the same dimensions?
+  ASSERT_EQ(table.size(), sampleSolution.size());
+  ASSERT_EQ(table.cols(), TABLE_WIDTH);
+
+  if (testForSorted) {
+    // Is the table sorted by join column?
+    auto oldEntry = table(0, jc);
+    for (size_t i = 1; i < table.size(); i++) {
+      ASSERT_TRUE(oldEntry <= table(i, jc));
+      oldEntry = table(i, jc);
+    }
   }
-  a.push_back({I(400000), I(200000)});
-  b.push_back({I(400000), I(200000)});
 
-  CALL_FIXED_SIZE_3(lwidth, rwidth, reswidth, J.join, a, 0, b, 0, &res);
-  ASSERT_EQ(6u, res.size());
+  // Sort both the table and the sampleSolution, so that both have a definite
+  // form for comparison.
+  auto sortFunction = [](
+      const auto& element1,
+      const auto& element2)
+    {
+      size_t i = 0;
+      while (i < (TABLE_WIDTH - 1) && element1[i] == element2[i]) {
+        i++;
+      }
+      return element1[i] < element2[i];
+    };
+  std::sort(table.begin(), table.end(), sortFunction);
+  std::sort(sampleSolution.begin(), sampleSolution.end(), sortFunction);
+
+  for (size_t row = 0; row < sampleSolution.size(); row++) {
+    for (size_t column = 0; column < TABLE_WIDTH; column++) {
+      ASSERT_EQ(I(sampleSolution[row][column]), table(row, column));
+    }
+  }
+}
 
-  a.clear();
-  b.clear();
-  res.clear();
+/*
+ * @brief Converts the given vectors to IdTables, joins them together with the
+ * given join function and returns the result.
+ *
+ * @tparam TABLE_A_WIDTH is the width of the first table.
+ * @tparam TABLE_B_WIDTH is the width of the second table.
+ * @tparam JOIN_FUNCTION is used to allow the transfer of any type of
+ *  lambda function, that could contain a join function. You never have to
+ *  actually specify this parameter , just let inference do its job.
+ *
+ * @param tableA, tableB hold the description of a table in form of vectors.
+ * @param jcA, jcB are the join columns for the tables.
+ * @param func the function, that will be used for joining the two tables
+ *  together. Look into src/engine/Join.h for how it should look like.
+ *
+ * @returns tableA and tableB joined together in a IdTable.
+ */
+template<size_t TABLE_A_WIDTH, size_t TABLE_B_WIDTH, typename JOIN_FUNCTION>
+IdTable useJoinFunctionOnVectorsTables(
+        const std::vector<std::array<size_t, TABLE_A_WIDTH>>& tableA,
+        const size_t jcA,
+        const std::vector<std::array<size_t, TABLE_B_WIDTH>>& tableB,
+        const size_t jcB,
+        JOIN_FUNCTION func,
+        ad_utility::source_location T = ad_utility::source_location::current()
```

Comment:
Don't call it `T`:)

Reply:
Done, see [29a2499](https://github.com/ad-freiburg/qlever/pull/821/commits/29a24994896c75f237b429704f2a70ea062bd561).

The variable shall henceforth be named `Franz_Friedrich_Feridnand_Von_Kruegen_The_Third`.